ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -980,8 +974,21 @@ object Serializers {
       readType().asInstanceOf[ReferenceType]
 
     def readPropertyName(): PropertyName = {
-      if (input.readBoolean()) readIdent()
-      else readTree().asInstanceOf[StringLiteral]
+      if (useHacks0614) {
+        if (input.readBoolean()) readIdent()
+        else readTree().asInstanceOf[StringLiteral]
+      } else {
+        input.readByte() match {
+          case TagPropertyNameIdent =>
+            readIdent()
+
+          case TagPropertyNameStringLiteral =>
+            readTree().asInstanceOf[StringLiteral]
+
+          case TagPropertyNameComputedName =>
+            ComputedName(readTree(), readString())
+        }
+      }
```

Comment:
I did review it quite thoroughly, but yes we should definitely try reading older IR.